kubernetes/samples/scripts/domain-lifecycle/helper.sh

```diff
@@ -0,0 +1,535 @@
+# !/bin/sh
+# Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+# Licensed under the Universal Permissive License v 1.0 as shown at https://oss.oracle.com/licenses/upl.
+#
+
+#
+# Function to get server start policy at cluster level
+# $1 - Domain resource in json format
+# $2 - Name of cluster
+# $3 - Return value containing cluster level server start policy
+#
+function getClusterPolicy {
+  local domainJson=$1
+  local clusterName=$2
+  local __clusterPolicy=$3
+  local effectivePolicy=""
+
+  clusterPolicyCmd="(.spec.clusters[] \
+    | select (.clusterName == \"${clusterName}\")).serverStartPolicy"
+  effectivePolicy=$(echo ${domainJson} | jq "${clusterPolicyCmd}")
+  eval $__clusterPolicy=${effectivePolicy}
+}
+
+#
+# Function to get server start policy at domain level
+# $1 - Domain resource in json format
+# $2 - Return value containing domain level server start policy
+#
+function getDomainPolicy {
+  local domainJson=$1
+  local __domainPolicy=$2
+  local effectivePolicy=""
+
+  clusterPolicyCmd=".spec.serverStartPolicy"
+  effectivePolicy=$(echo ${domainJson} | jq "${clusterPolicyCmd}")
+  eval $__domainPolicy=${effectivePolicy}
+}
+
+#
+# Function to get effective start policy of server
+# $1 - Domain resource in json format
+# $2 - Name of server
+# $3 - Name of cluster
+# $4 - Return value containing effective server start policy
+#
+function getEffectivePolicy {
+  local domainJson=$1
+  local serverName=$2
+  local clusterName=$3
+  local __currentPolicy=$4
+  local currentPolicy=""
+
+  getServerPolicy "${domainJson}" "${serverName}" currentPolicy
+  if [[ -z "${currentPolicy}" ||  ${currentPolicy} == "null" ]]; then
```

Comment:
slight simplification: would it help to  have the 'get' helpers  convert 'null' ret vals to '', so you don't need to check for both "null" and -z everywhere?